instrumentation/graphql/lib/opentelemetry/instrumentation/graphql/patches/opentelemetry_graphql_tracing.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module GraphQL
+      module Patches
+        # OpenTelemery Graphql platform_tracing implementation for instrumentation
+        class OpenTelemetryGraphQLTracing < ::GraphQL::Tracing::PlatformTracing
+          self.platform_keys = {
+            'lex' => 'lex.graphql',
+            'parse' => 'parse.graphql',
+            'validate' => 'validate.graphql',
+            'analyze_query' => 'analyze.graphql',
+            'analyze_multiplex' => 'analyze.graphql',
+            'execute_multiplex' => 'execute.graphql',
+            'execute_query' => 'execute.graphql',
+            'execute_query_lazy' => 'execute.graphql'
+          }
+
+          def platform_trace(platform_key, key, data)
+            tracer.in_span(
+              platform_key,
+              attributes: {},
+              kind: :client
+            ) do |span|
+              if key == 'execute_multiplex'
+                operations = data[:multiplex].queries.map(&:selected_operation_name).join(', ')
+                span.set_attribute('operation', operations)
+              end
+
+              if key == 'execute_query'
+                span.set_attribute(:selected_operation_name, data[:query].selected_operation_name)
+                span.set_attribute(:selected_operation_type, data[:query].selected_operation.operation_type)
+                span.set_attribute(:query_string, data[:query].query_string)
+              end
+              yield
+            end
+          end
+
+          def tracer
+            GraphQL::Instrumentation.instance.tracer
+          end
+
+          def platform_field_key(type, field)
+            "#{type.graphql_name}.#{field.graphql_name}"
```

Comment:
From our SIG call today, this is the key that turned an 11 span trace into a 2500 span trace.  However I really do think the information that comes from this field is valuable.  

I'm uncertain as to whether or not this should be on by default, I know for us (Shopify) we would want to configure this to be off.